CHANGELOG.md

```diff
@@ -0,0 +1,477 @@
+
+## Version 2.5.1 (May 1, 2019)
+
+-   [JENKINS-57257](https://issues.jenkins-ci.org/browse/JENKINS-57257): Fixed regression in 2.5.0 that caused for PRs 
+    from forks to fail
+
+## Version 2.5.0 (April 26, 2019)
+
+#### Feature
+
+  Today, any PR that requires merging ends up cloning the associated GitHub repository on a Jenkins master, 
+  regardless of whether the Jenkinsfile was changed or not. In this release, we have significantly reduced 
+  the number of occurrences in which unnecessary full clones on the master happen.
+    
+  For details see: [JENKINS-43194](https://issues.jenkins-ci.org/browse/JENKINS-43194) - Lightweight checkout 
+  for PR merge jobs.
+
+## Version 2.4.5 (March 27, 2019)
+
+-   Doc fixes: [\#194](https://github.com/jenkinsci/github-branch-source-plugin/pull/194), [\#184](https://github.com/jenkinsci/github-branch-source-plugin/pull/184), [\#196](https://github.com/jenkinsci/github-branch-source-plugin/pull/196), [\#201](https://github.com/jenkinsci/github-branch-source-plugin/pull/201)
+-   [JENKINS-45504](https://issues.jenkins-ci.org/browse/JENKINS-45504): Add @Symbol annotations to traits 
+    ([\#183](https://github.com/jenkinsci/github-branch-source-plugin/pull/183)
+-   [JENKINS-44715](https://issues.jenkins-ci.org/browse/JENKINS-44715): Pipeline plugin won't get updated PR github 
+    title [\#192](https://github.com/jenkinsci/github-branch-source-plugin/pull/192)
+-   Fixed a potential problem when discovering mixed case Jenkinsfile 
+    [\#198](https://github.com/jenkinsci/github-branch-source-plugin/pull/198)
+-   [JENKINS-54126](https://issues.jenkins-ci.org/browse/JENKINS-54126) Diagnostics; surface a potential cause to the 
+    scan log [\#200](https://github.com/jenkinsci/github-branch-source-plugin/pull/200)
+-   List github orgs [\#187](https://github.com/jenkinsci/github-branch-source-plugin/pull/187)
+-   [JENKINS-46094](https://issues.jenkins-ci.org/browse/JENKINS-46094): "Checkout over SSH" fails the build since it 
+    still uses HTTPS urls [\#205](https://github.com/jenkinsci/github-branch-source-plugin/pull/205)
+-   [JENKINS-54051](https://issues.jenkins-ci.org/browse/JENKINS-54051): GitHub-Branch-Source plugin 2.3.5 Security 
+    Update causing error in adding GitHub Enterprise Servers [\#210](https://github.com/jenkinsci/github-branch-source-plugin/pull/210)
+
+## Version 2.4.2 (Jan 16, 2019)
+
+-   [JENKINS-52397](https://issues.jenkins-ci.org/browse/JENKINS-52397): Org Scan blows up when repository has no tags [\#191](https://github.com/jenkinsci/github-branch-source-plugin/pull/191)
+-   [INFRA-1934](https://issues.jenkins-ci.org/browse/INFRA-1934): Stop publishing to jenkinsci/jenkins repo on Docker Hub 
+
+## Version 2.4.1 (Oct 15, 2018)
+
+-   [JENKINS-54046](https://issues.jenkins-ci.org/browse/JENKINS-54046): Disabled the cache by default for Windows masters.
+
+## Version 2.4.0 (Oct 4, 2018) - Version 2.4.0-beta-1 (Oct 2, 2018)
```

Comment:
This seems a little awkward, and it's awkward in the wiki too. I'm going to delete the stuff about the beta, since that seems pretty unimportant now that it's been a year and change (and the releases were two days apart).